lib/MLDataDevices/src/public.jl

```diff
@@ -9,8 +9,10 @@ end
 struct MetalDevice <: AbstractGPUDevice end
 struct oneAPIDevice <: AbstractGPUDevice end
 
-# TODO: Later we might want to add the client field here?
-struct XLADevice <: AbstractAcceleratorDevice end
+@kwdef struct XLADevice{C, D} <: AbstractAcceleratorDevice
```

Comment:
For future extensibility [as we get distributed and other things setup], can this be renamed to ReactantDevice?

Reply:
yeah I can do it with a `const XLADevice = ReactantDevice` to preserve backward compatibility.

Reply:
Sounds good!